### PR TITLE
Add support for rerunning unsafe queries

### PR DESCRIPTION
**Pull request by elshize**

### CODE REVIEW

File: tools/queries.cpp [hunk 1]
```diff
@@ -60,16 +60,25 @@ void op_perftest(Functor query_func,
...
+    std::size_t num_reruns = 0;
+    spdlog::info("Safe: {}", safe);
 
     for (size_t run = 0; run <= runs; ++run) {
         size_t idx = 0;
         for (auto const &query : queries) {
             auto usecs = run_with_timer<std::chrono::microseconds>([&]() {
                 uint64_t result = query_func(query, thresholds[idx]);
+                if (safe && result < k) {
+                    spdlog::warn("results ({}) < k ({})", result, k);
```
Comment (amallia) [inline tools/queries.cpp]: We are timing this. Can we drop this log line?

File: tools/queries.cpp [hunk 2]
```diff
@@ -152,7 +163,7 @@ void perftest(const std::string &index_filename,
         std::ifstream tin(*thresholds_filename);
         size_t idx = 0;
         while (std::getline(tin, t)) {
-            thresholds[idx] = std::stof(t);
+            thresholds[idx] = std::stof(t) - 0.0001;
```
Comment (amallia) [inline tools/queries.cpp]: Maybe we should fix the thresholds instead. 
I propose to get rid of `- 0.0001` here and fix the way we compute thresholds in a separate PR.

**Review by amallia — CHANGES_REQUESTED** [review]
Looks good. I would like to change just a couple of things

**Review by amallia — APPROVED** [review]

**Review by JMMackenzie — CHANGES_REQUESTED** [review]
If you don't pass any thresholds then `safe` should remain false. Otherwise any query with fewer than `k` results will run twice for no reason, right?

**Review by JMMackenzie — APPROVED** [review]

**Pull request merged**